script.js

```diff
@@ -0,0 +1,328 @@
+    } else if (event.toElement.id === 'CapsLock'
+      && document.querySelector(`#${array[i][2]}`).classList.contains('upperCase') === true) {
+      document.getElementById(array[i][2]).innerHTML = document.getElementById(array[i][2]).innerHTML.toLowerCase();
+      document.querySelector(`#${array[i][2]}`).classList.remove('upperCase');
+      document.querySelector('#CapsLock').classList.remove('changeCaps');
+    }
+  }
+});
+
+// make the Back key
+document.addEventListener('keydown', (event) => {
+  if (event.code === 'Backspace') {
+    const output = document.getElementById('textarea');
+    output.value = output.value.substring(0, output.value.length - 1);
+  }
+});
+
+document.addEventListener('click', (event) => {
+  if (event.toElement.id === 'Backspace') {
+    const output = document.getElementById('textarea');
+    output.value = output.value.substring(0, output.value.length - 1);
+  }
+});
+
+// make the Space key
+document.addEventListener('keydown', (event) => {
+  if (event.code === 'Space') {
+    document.getElementById('textarea').value = `${document.getElementById('textarea').value} `;
+  }
+});
+
+document.addEventListener('click', (event) => {
+  if (event.toElement.id === 'Space') {
+    document.getElementById('textarea').value = `${document.getElementById('textarea').value} `;
+  }
+});
+
+// make the Tab key
+document.addEventListener('keydown', (event) => {
+  if (event.code === 'Tab') {
+    document.getElementById('textarea').value = `${document.getElementById('textarea').value}    `;
+  }
+});
+
+document.addEventListener('click', (event) => {
+  if (event.toElement.id === 'Tab') {
+    document.getElementById('textarea').value = `${document.getElementById('textarea').value}    `;
+  }
+});
+
+// make the Enter key
+document.addEventListener('keydown', (event) => {
+  if (event.code === 'Enter') {
+    document.getElementById('textarea').value = `${document.getElementById('textarea').value}\n`;
+  }
+});
+
+document.addEventListener('click', (event) => {
+  if (event.toElement.id === 'Enter') {
+    document.getElementById('textarea').value = `${document.getElementById('textarea').value}\n`;
+  }
+});
+
+// make the Shift key
+document.addEventListener('keydown', (event) => {
+  for (let i = 0; i < array.length; i += 1) {
+    if (event.code === 'ShiftLeft'
+      && i !== 13 && i !== 14 && i !== 27 && i !== 28 && i !== 41
+      && i !== 42 && i !== 54 && i !== 55 && i !== 56 && i !== 57 && i !== 58 && i !== 59 && i !== 63) {
+      document.getElementById(array[i][2]).innerHTML = document.getElementById(array[i][2]).innerHTML.toUpperCase();
+      if (i === 0 || i === 1 || i === 2 || i === 3 || i === 4 || i === 5 || i === 6
+        || i === 7 || i === 8 || i === 9 || i === 10 || i === 11 || i === 12 || i === 25
+        || i === 26 || i === 38 || i === 39 || i === 40 || i === 50 || i === 51 || i === 52) {
+        if (document.getElementById(array[49][2]).innerHTML === 'm'
```

Comment:
makes sense to extract to const `document.getElementById(array[49][2]` instead of searching it 16 times